[SLES][RHEL8] error: no member named 'exclusive_scan' in namespace 'std

https://github.com/ROCm/MIOpen/blob/upgrade_environ/src/include/miopen/graphapi/util.hpp#L70

sles and rhel8 build failed, 
```
2024-06-11T10:13:11.549602605Z In file included from /long_pathname_so_that_rpms_can_package_the_debug_info/src/extlibs/MLOpen/src/graphapi/find_engine.cpp:36:
2024-06-11T10:13:11.549641496Z /long_pathname_so_that_rpms_can_package_the_debug_info/src/extlibs/MLOpen/src/include/miopen/graphapi/util.hpp:70:10: error: no member named 'exclusive_scan' in namespace 'std'
2024-06-11T10:13:11.549648076Z    70 |     std::exclusive_scan(dims.begin(), dims.end(), strides.begin(), T{1LL}, std::multiplies<T>{});
2024-06-11T10:13:11.549658586Z   
```

--- a/src/include/miopen/graphapi/util.hpp
+++ b/src/include/miopen/graphapi/util.hpp
@@ -67,7 +67,19 @@ Tensor makeTensor(std::string_view name, miopenDataType_t dt, const Vec& dims)
 
     Vec strides(dims);
     using T = typename Vec::value_type;
-    std::exclusive_scan(dims.begin(), dims.end(), strides.begin(), T{1LL}, std::multiplies<T>{});
+    // former std::exclusive_scan(dims.begin(),
+    //                            dims.end(),
+    //                            strides.begin(),
+    //                            T{1LL}, std::multiplies<T>{});
+    // replaced by the loop due to std::exclusive_scan is not supported on gcc older that 11
+    if(!dims.empty())
+    {
+        strides[0] = static_cast<T>(1);
+        for(size_t i = 0; i < dims.size() - 1; ++i)
+        {
+            strides[i + 1] = dims[i] * strides[i];
+        }
+    }
 
     return makeTensor<isVirtual>(name, dt, dims, strides);
 }

